src/Common/DNSResolver.cpp

```diff
@@ -138,16 +140,10 @@ static DNSResolver::IPAddresses resolveIPAddressImpl(const std::string & host)
     return addresses;
 }
 
-static String reverseResolveImpl(const Poco::Net::IPAddress & address)
+static std::vector<String> reverseResolveImpl(const Poco::Net::IPAddress & address)
 {
-    Poco::Net::SocketAddress sock_addr(address, 0);
-
-    /// Resolve by hand, because Poco::Net::DNS::hostByAddress(...) does getaddrinfo(...) after getnameinfo(...)
-    char host[1024];
-    int err = getnameinfo(sock_addr.addr(), sock_addr.length(), host, sizeof(host), nullptr, 0, NI_NAMEREQD);
-    if (err)
-        throw Exception("Cannot getnameinfo(" + address.toString() + "): " + gai_strerror(err), ErrorCodes::DNS_ERROR);
-    return host;
+    auto ptr_resolver = DB::DNSPTRResolverProvider::get();
```

Comment:
A comment is required. It should explain why `getaddrinfo` and `gethostbyaddr_r` are not suitable.

Reply:
I have added that in `CaresPTRResolver` docs. Please let me know if that's enough